app/lib/analyzer/backend.dart

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) 2017, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:math';
+
+import 'package:gcloud/db.dart';
+
+import '../shared/utils.dart';
+
+import 'models.dart';
+
+final Random _random = new Random.secure();
+
+/// Datastore-related access methods for the analyzer service
+class AnalysisBackend {
+  final DatastoreDB db;
+
+  AnalysisBackend(this.db);
+
+  /// Gets the [Analysis] object from the Datastore. [version] and [analysis] is
+  /// optional, when they are missing the analysis from the latest rollup will
+  /// be returned.
+  Future<Analysis> getAnalysis(String package,
+      [String version, int analysis]) async {
+    final Key packageKey = db.emptyKey.append(PackageAnalysis, id: package);
+
+    if (version == null) {
+      final list = await db.lookup([packageKey]);
+      final PackageAnalysis pa = list[0];
+      if (pa == null) return null;
+      version = pa.latestVersion;
+      analysis = pa.latestAnalysis;
```

Comment:
There's no `PackageAnalysis.latestAnalysis` now I think. Just remove this line.

(Since the travis bot didn't catch it and the variable is typed, I assume we're not running the analyzer on travis. We should keep that in mind and enable it at some point.)